src/main/java/uk/gov/hmcts/reform/sendletter/tasks/UploadLettersTask.java

```diff
@@ -79,20 +84,25 @@ public void run() {
     }
 
     private void uploadToFtp(Letter letter) {
-        FileToSend file = new FileToSend(
-            FinalPackageFileNameHelper.generateName(letter),
-            letter.getFileContent()
-        );
-
-        ftp.upload(file, isSmokeTest(letter), letter.getService());
-
-        logger.info(
-            "Uploaded letter id: {}, messageId: {}, file name: {}, additional data: {}",
-            letter.getId(),
-            letter.getMessageId(),
-            file.filename,
-            letter.getAdditionalData()
-        );
+        Optional<String> serviceFolder = serviceFolderMapping.getFolderFor(letter.getService());
+        if (serviceFolder.isPresent()) {
+            FileToSend file = new FileToSend(
+                FinalPackageFileNameHelper.generateName(letter),
+                letter.getFileContent()
+            );
+
+            ftp.upload(file, isSmokeTest(letter), serviceFolder.get());
+
+            logger.info(
+                "Uploaded letter id: {}, messageId: {}, file name: {}, additional data: {}",
+                letter.getId(),
+                letter.getMessageId(),
+                file.filename,
+                letter.getAdditionalData()
```

Comment:
Is additional data not sensitive information?

Reply:
Those should claim reference or ccd case reference and it helps for debugging and analysing issues on prod. Services should not send any sensitive data afaik :)